test/PoolManager.t.sol

```diff
@@ -808,6 +808,99 @@ contract PoolManagerTest is Test, Deployers, GasSnapshot {
     }
 
+    function test_swap_toLiquidity_fromMinPrice() public {
+        PoolKey memory _key = PoolKey(currency0, currency1, 500, 10, IHooks(address(0)));
+        manager.initialize(_key, TickMath.MIN_SQRT_PRICE);
+
+        IPoolManager.ModifyLiquidityParams memory params =
+            IPoolManager.ModifyLiquidityParams({tickLower: -10, tickUpper: 10, liquidityDelta: 100e18, salt: 0});
+        modifyLiquidityRouter.modifyLiquidity(_key, params, ZERO_BYTES);
+
+        // zeroForOne=false to swap higher
+        IPoolManager.SwapParams memory swapParams = IPoolManager.SwapParams(false, -1, TickMath.MAX_SQRT_PRICE - 1);
+        PoolSwapTest.TestSettings memory testSettings =
+            PoolSwapTest.TestSettings({takeClaims: false, settleUsingBurn: false});
+
+        (uint160 sqrtPriceX96,,,) = manager.getSlot0(_key.toId());
+
+        swapRouter.swap(_key, swapParams, testSettings, ZERO_BYTES);
+
+        (sqrtPriceX96,,,) = manager.getSlot0(_key.toId());
+
+        // The swap pushes the price to the start of liquidity.
+        assertEq(sqrtPriceX96, TickMath.getSqrtPriceAtTick(-10));
+    }
+
+    function test_fuzz_swap_toLiquidity(uint160 sqrtPriceX96) public {
+        PoolKey memory _key = PoolKey(currency0, currency1, 500, 10, IHooks(address(0)));
+        sqrtPriceX96 = uint160(bound(sqrtPriceX96, TickMath.MIN_SQRT_PRICE, TickMath.MAX_SQRT_PRICE - 1));
+        vm.assume(sqrtPriceX96 != 0);
+        manager.initialize(_key, sqrtPriceX96);
+
+        int24 tick = TickMath.getTickAtSqrtPrice(sqrtPriceX96);
+
+        bool zeroForOne;
+        uint160 sqrtPriceX96Limit;
+        int24 tickBoundary;
+        if (tick <= -10) {
+            // zeroForOne=false to swap higher
+            zeroForOne = false;
+            sqrtPriceX96Limit = TickMath.MAX_SQRT_PRICE - 1;
+            tickBoundary = -10;
+        } else if (tick > 10) {
+            // zeroForOne=true to swap lower
+            zeroForOne = true;
+            sqrtPriceX96Limit = TickMath.MIN_SQRT_PRICE + 1;
+            tickBoundary = 9;
+        } else {
+            // the price is between the position, so let's just swap down
+            zeroForOne = true;
+            sqrtPriceX96Limit = TickMath.MIN_SQRT_PRICE + 1;
+            // the tick will just stay the same, we're only swapping 1 wei
+            tickBoundary = TickMath.getTickAtSqrtPrice(sqrtPriceX96);
+        }
+
+        // deeeeeep liquidity so that swapping 1 wei doesn't change the price too much if the price is within the tick range
+        IPoolManager.ModifyLiquidityParams memory params = IPoolManager.ModifyLiquidityParams({
+            tickLower: -10,
+            tickUpper: 10,
+            liquidityDelta: 100000000000e18,
+            salt: 0
+        });
+
+        modifyLiquidityRouter.modifyLiquidity(_key, params, ZERO_BYTES);
+
+        IPoolManager.SwapParams memory swapParams = IPoolManager.SwapParams(zeroForOne, 1, sqrtPriceX96Limit);
```

Comment:
The failure happens because the price does move slightly. It's not ideal for the test to fail probabilistically. It [fails](https://github.com/Uniswap/v4-core/actions/runs/11377758111/job/31652432976?pr=904) because the `getTickAtSqrtPrice` flips from `0` to `-1`.

I think that case would pass if this `amountSpecified` was `-1`, because that amount would become fee and not move the price.

Reply:
Yeah I meant to change to -1 thanks!